### PR TITLE
Restructure docs and split into versioned and unversioned

### PR DESCRIPTION
**Pull request by youngnick**
Signed-off-by: Nick Young <ynick@vmware.com>

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:

This document updates the docs site structure to have less things versioned, and adds top-level navigation. Basically, only the API Types, Guides and API reference are versioned now, since I don't think  we've been changing the other things along version boundaries.

I've tried to make the navigation make a bit more sense, but haven't touched the content of the files too much (although I did notice heaps of things to fix, I'll keep those for followups).

This is a first cut, intended to set us up to be in the best place we can be without substantially changing the docs flow (by adding `mike` into the mkdocs flow, moving to Hugo, or other big change).

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by youngnick** [comment]
/hold

To allow multiple people to review.

**Comment by youngnick** [comment]
https://deploy-preview-782--kubernetes-sigs-gateway-api.netlify.app/ for previewing.

File: mkdocs.yml [hunk 1]
```diff
@@ -22,32 +24,49 @@ markdown_extensions:
...
+      - API Types:
+          GatewayClass: v1alpha2/api-types/gatewayclass.md
+          Gateway: v1alpha2/api-types/gateway.md
+          HTTPRoute: v1alpha2/api-types/httproute.md
+      - Guides:
+        - Getting started: v1alpha2/guides/getting-started.md
+        - Simple Gateway: v1alpha2/guides/simple-gateway.md
+        - HTTP routing: v1alpha2/guides/http-routing.md
+        - HTTP traffic splitting: v1alpha2/guides/traffic-splitting.md
+        - Multiple namespaces and routes: v1alpha2/guides/multiple-ns.md
```
Comment (robscott) [inline mkdocs.yml:45]: Most of these are out of date as captured by #783, until we update them, do you mind adding a warning on those pages similar to what I had in #778? `!!! danger "This page has not been updated for v1alpha2 yet."`

**Review by robscott — COMMENTED** [review]
This is a huge improvement! Thanks @youngnick!

File: mkdocs.yml [hunk 2]
```diff
@@ -22,32 +24,49 @@ markdown_extensions:
...
-  - References:
-    - API specification: references/spec.md
-    - Releases: references/releases.md
-    - Implementations: references/implementations.md
-  - ... | geps/gep-*.md
+  - Overview:
+    - Introduction: index.md
+    - Concepts:
+        API overview: concepts/api-overview.md
+        Security Model: concepts/security-model.md
```
[the remark this answers is not in the document]
Comment (youngnick) [reply mkdocs.yml:31]: Okay, I'll mark this with a "Needs update" danger note as well.

**Comment by youngnick** [comment]
I folded in the above suggestions plus the ones from the call, ready for a final (hopefully!) look.

**Comment by robscott** [comment]
Looks great, thanks @youngnick!

/lgtm

**Comment by robscott** [comment]
I think it's probably safe to remove the hold on this since it seems like we reached consensus on this approach during the community meeting earlier today. If no one objects, I'll remove the hold in a few hours.

**Pull request merged**